### PR TITLE
tikv: support dynamic modifications on `raftstore.store-io-pool-size`

### PR DESCRIPTION
**Pull request by LykxSassinator**
Signed-off-by: Lucasliang <nkcs_lykx@hotmail.com>

### First-time contributors' checklist 

- [x] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### What is changed, added or deleted? (Required)

Add docs on the dynamic modification supports to `raftstore.store-io-pool-size`.

Ref: https://github.com/tikv/tikv/issues/13964

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [x] v6.6 (TiDB 6.6 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.4 (TiDB 6.4 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

**Comment by TomShawn** [comment]
@LykxSassinator Could you invite a technical review? Thanks

**Comment by LykxSassinator** [comment]
> @LykxSassinator Could you invite a technical review? Thanks

Get. Related PR on TiKV needs to be polished, once it finished will I add related reviewers to review it again.

**Comment by LykxSassinator** [comment]
@tonyxuqqi After https://github.com/tikv/tikv/pull/13965 merged, PTAL, thx.

**Review by shichun-0415 — APPROVED** [review]
LGTM

**Review by Oreoxmt — APPROVED** [review]
LGTM

File: dynamic-config.md [hunk 1]
```diff
@@ -156,6 +156,7 @@ show warnings;
 | raftstore.store-pool-size | 处理 Raft 的线程池中线程的数量，即 Raftstore 线程池的大小 |
 | raftstore.apply-max-batch-size | Raft 状态机由 BatchSystem 批量执行数据写入请求，该配置项指定每批可执行请求的最多 Raft 状态机个数。 |
 | raftstore.store-max-batch-size |  Raft 状态机由 BatchSystem 批量执行把日志落盘至磁盘的请求，该配置项指定每批可执行请求的最多 Raft 状态机个数。 |
+| raftstore.store-io-pool-size | 处理 Raft I/O 任务的线程池中线程的数量，即 StoreWriter 线程池的大小（不支持将该配置项由非 0 值调整为 0，或者从 0 调整为非 0 值）|
```
Comment (Oreoxmt) [inline dynamic-config.md:159]:
```suggestion
| raftstore.store-io-pool-size | 处理 Raft I/O 任务的线程池中线程的数量，即 StoreWriter 线程池的大小（不支持将该配置项由非零值调整为 0，或者从 0 调整为非零值）|
```

**Review by Oreoxmt — APPROVED** [review]
Rest LGTM

**Review by Oreoxmt — APPROVED** [review]

**Comment by LykxSassinator** [comment]
@Oreoxmt cc, this PR can be merged, as https://github.com/tikv/tikv/pull/13965 has been merged.

**Pull request merged**